Tests/BotTests/GitHub/GitHubAPITests.swift

```diff
@@ -0,0 +1,286 @@
+                source: .init(ref: "fix_fmtsort", sha: "57a206bf54e4242af1eae01b7b351f332c425c2f"),
+                target: .init(ref: "master", sha: "919594830f17f25c9e971934d825615463ad8a10"),
+                labels: [.init(name: "cla: yes")]
+            )
+        }
+    }
+
+    func test_fetch_pull_request_number() {
+
+        perform(
+            stub: Interceptor.loadOrRecordStubs(into: directory.appendingPathComponent("fetch_pull_request_number.json"))
+        ) { client in
+
+                let api = RepositoryAPI(client: client, repository: .init(owner: "golang", name: "go"))
+
+                let result = api.fetchPullRequest(number: 33248).first()?.value
+
+                expect(result).toNot(beNil())
+                expect(result) == PullRequestMetadata(
+                    reference: PullRequest(
+                        number: 33248,
+                        title: "runtime: fix gdb pretty print for slices",
+                        author: .init(login: "elbeardmorez"),
+                        source: .init(ref: "gdb_print_slice_fix", sha: "6e12bd85f5d71569cbfe574612210d3c925881b7"),
+                        target: .init(ref: "master", sha: "e8c7e639ea6f4e2c66d8b17ca9283dba53667c9d"),
+                        labels: [.init(name: "cla: yes")]
+                    ),
+                    isMerged: false,
+                    mergeState: .clean
+                )
+        }
+    }
+
+    func test_fetch_commit_status() {
+        perform(
+            stub: Interceptor.loadOrRecordStubs(into: directory.appendingPathComponent("fetch_commit_status.json"))
+        ) { client in
+
+            let api = RepositoryAPI(client: client, repository: .init(owner: "golang", name: "go"))
+
+            let result = api.fetchCommitStatus(for: target).first()?.value
+
+            expect(result).toNot(beNil())
+            expect(result) == CommitState(
+                state: .success,
+                statuses: [
+                    CommitState.Statuses(
+                        state: .success,
+                        description: "All necessary CLAs are signed",
+                        context: "cla/google"
+                    )
+                ]
+            )
+        }
+    }
+
+    func test_delete_branch() {
+
+        perform(stub:
+            Interceptor.load(
+                stubs: [
+                    Interceptor.Stub(
+                        response: Interceptor.Stub.Response(
+                            url: URL(string: "https://api.github.com/repos/golang/go/git/refs/heads/gdb_print_slice_fix")!,
+                            statusCode: 204,
+                            body: Data()
+                        )
+                    )]
+            )
+        ) { client in
+
+            let api = RepositoryAPI(client: client, repository: .init(owner: "golang", name: "go"))
+
+            let result: Void? = api.deleteBranch(named: target.source).first()?.value
+
+            expect(result).toNot(beNil())
```

Comment:
Should we also have a test checking when `deleteBranch` (and all other similar tests) returns an error? (Or are error responses handled in a more global test that I missed?)

Reply:
You're right we don't have them but I'll create an issue to add them but I'd like to not extend this one even more as this can be handled separately.

Reply:
Just asking to be sure you created the ticket (nit: if so it can be nice to add a `// [CNSMR-xxx] Add test for failure cases` comment somewhere here to reference it btw)